bundles/framework/maplegend/Flyout.js

```diff
@@ -100,175 +105,47 @@ Oskari.clazz.define('Oskari.mapframework.bundle.maplegend.Flyout',
-                        sandbox.postRequestByName('catalogue.ShowMetadataRequest', [{
-                            uuid: uuid
-                        }]);
+                const layerLegendImage = layer.getLegendImage ? layer.getLegendImage() : null;
+
+                if (layerLegendImage) {
+                    titles.push({
+                        title: Oskari.util.sanitize(layer.getName()),
+                        uuid: uuid,
+                        legendImageURL: layerLegendImage || null,
+                        loadError: false,
+                        showMetadataCallback: layerLegendImage ? (event, layerUuid) => this.showMetadataFlyout(event, layerUuid) : null
                     });
                 }
-
-                if (layerContainer !== null) {
-                    accordionPanel = Oskari.clazz.create('Oskari.userinterface.component.AccordionPanel');
-                    accordionPanel.open();
-                    accordionPanel.setTitle(layerTitle);
-                    accordionPanel.getContainer().append(layerContainer);
-                    accordion.addPanel(accordionPanel);
-                }
             }
 
-            // If no legend images at all, inform the user
-            me._informNoLegendImages();
-        },
-
-        /**
-         * @method _createLayerContainer
-         * @private
-         * Creates the layer containers
-         * @param {Oskari.mapframework.domain.WmsLayer/Oskari.mapframework.domain.WfsLayer/Oskari.mapframework.domain.VectorLayer/Object} layer to render
-         */
-        _createLayerContainer: function (layer) {
-            var me = this,
-                layerDiv = this.templateLayer.clone();
-
-            /* let's not show same image multiple times */
-            var imagesAdded = {};
-            /* main layer div */
-            var legendDiv = me._createLegendDiv(layer, imagesAdded);
-            if (legendDiv && legendDiv !== null) {
-                layerDiv.append(legendDiv);
-
-                /* optional sublayers */
-                var sublayers = layer.getSubLayers ? layer.getSubLayers() : null,
-                    sl,
-                    sublayer,
-                    subLayerlegendDiv;
-
-                if (sublayers) {
-                    for (sl = 0; sl < sublayers.length; sl += 1) {
-                        sublayer = sublayers[sl];
-                        subLayerlegendDiv = me._createLegendDiv(sublayer, imagesAdded);
-                        if (!subLayerlegendDiv && subLayerlegendDiv !== null) {
-                            continue;
-                        }
-                        layerDiv.append(subLayerlegendDiv);
+            ReactDOM.render(
+                <LocaleProvider value={{ bundleKey: 'maplegend' }}>
+                    { titles.length === 0
+                        ? <Message messageKey='noLegendsText' />
+                        : <MapLegendList legendList={ titles } noImageText={ <Message messageKey='invalidLegendUrl' /> } />
```

Comment:
I think we can bake the `noImageText` in `MapLegendList` instead of giving it as prop.